nxp_s32k/libuavcan/s32k_libuavcan.hpp

```diff
@@ -0,0 +1,958 @@
+            PTE->PCOR |= 1 << 11;              /* Set the pin LOW */
+
+            PORTE->PCR[10] |= PORT_PCR_MUX(1); /* Same as above */
+            PTE->PDDR |= 1 << 10;
+            PTE->PCOR |= 1 << 10;
+        }
+
+#endif
+
+#if defined(MCU_S32K148)
+        PCC->PCCn[PCC_PORTB_INDEX] |= PCC_PCCn_CGC_MASK; /* Clock gating to PORT B */
+        PORTB->PCR[12] |= PORT_PCR_MUX(4);               /* CAN2_RX at PORT B pin 12 */
+        PORTB->PCR[13] |= PORT_PCR_MUX(4);               /* CAN2_TX at PORT B pin 13 */
+#endif
+
+        /* If function ended successfully, return address of object member of type S32K_InterfaceGroup */
+        out_group = &S32K_InterfaceGroupObj_;
+
+        /* Return code for start of S32K_InterfaceGroup */
+        return Status;
+    }
+
+    /**
+     * Deinitializes the peripherals needed for the current libuavcan driver layer.
+     * @param inout_group Pointer that will be set to null
+     * @return libuavcan::Result::Success. If the used peripherals were deinitialized properly.
+     */
+    virtual libuavcan::Result stopInterfaceGroup(InterfaceGroupPtrType& inout_group) override
+    {
+        /* Initialize return value status */
+        libuavcan::Result Status = libuavcan::Result::Success;
+
+        /* FlexCAN module deinitialization */
+        for (std::uint8_t i = 0; i < S32K::CANFD_Count; i++)
+        {
+            /* Disable FlexCAN module */
+            S32K::FlexCAN[i]->MCR |= CAN_MCR_MDIS_MASK;
+
+            if (isSuccess(Status))
+            {
+                /* Poll for Low Power ACK, waits for current transmission/reception to finish */
+                Status = S32K::flagPollTimeout_Set(S32K::FlexCAN[i]->MCR, CAN_MCR_LPMACK_MASK);
+
+                if (isSuccess(Status))
+                {
+                    /* Disable FlexCAN clock gating */
+                    PCC->PCCn[S32K::PCC_FlexCAN_Index[i]] &= ~PCC_PCCn_CGC_MASK;
+                }
+            }
+        }
+
+        /* Assign to null the pointer output argument */
+        inout_group = nullptr;
+
+        /* Return status code of successful stop of S32K_InterfaceGroup */
+        return Status;
+    }
+
+    /**
+     * Return the number of filters that the current UAVCAN node can support.
+     * @return The maximum number of frame filters available for filter groups managed by this object,
+     *         i.e. the number of combinations of ID and mask that each FlexCAN instance supports
+     */
+    virtual std::size_t getMaxFrameFilters() const override { return S32K::Filter_Count; }
+};
+
+}  // END namespace media
+}  // END namespace libuavcan
+
+/**
+ * Interrupt requests handled by hardware in each frame reception installed by the linker
+ * in function of the number of instances available in the target MCU
+ */
+extern "C"
+{
+    void CAN0_ORed_0_15_MB_IRQHandler() { libuavcan::media::S32K_InterfaceGroup::S32K_libuavcan_ISR_handler(0u); }
```

Comment:
Add `__attribute__((interrupt("IRQ")))` since this is an ISR.

Your demo will not work, by-the-way, because of this. You'll have the wrong calling convention and the `instance` variable provided to the "ISR handler" will be garbage which will cause you to index into an undefined memory location.